src/prime.py

```diff
@@ -42,13 +44,15 @@ def find_prime(template, parallel_strategy=(1, 0)):
         for (i, d) in enumerate(template)
         if d == -1
     ]
-    for digits in combinations_with_replacement(range(10), len(variables)):
+    # TODO: combinations_with_replacement is wrong (e.g. [11, 12, 22] -- does not include 21)
+    options = itertools.combinations_with_replacement(range(10), len(variables))
+    for i, digits in enumerate(options):
+        if i % n != r:
+            continue
```

Comment:
This is a better way to break up work (1) since it doesn't require editing the template and (2) since it results in more even workloads.